src/Phinx/Migration/Manager.php

```diff
@@ -134,10 +134,11 @@ public function printStatus($environment, $format = null)
      *
      * @param string    $environment Environment
      * @param \DateTime $dateTime    Date to migrate to
+     * @param boolean   $dryRun      Output SQL rather than executing it
      *
      * @return void
      */
-    public function migrateToDateTime($environment, \DateTime $dateTime)
+    public function migrateToDateTime($environment, \DateTime $dateTime, $dryRun)
```

Comment:
I think the input was excluded from the manager deliberately to isolate coupling of it to the command classes. The way it's presently written, all needed console arguments are [extracted at the command class level](https://github.com/robmorgan/phinx/blob/a601da707f00d40749fe8b2439c28368ed06daed/src/Phinx/Console/Command/Migrate.php#L74-L76) and then passed on.

Manager is [coupled to config](https://github.com/robmorgan/phinx/blob/a601da707f00d40749fe8b2439c28368ed06daed/src/Phinx/Migration/Manager.php#L63). That [implements `ArrayAccess`](https://github.com/robmorgan/phinx/blob/a601da707f00d40749fe8b2439c28368ed06daed/src/Phinx/Config/ConfigInterface.php#L37) and as such is mutable. Commands are [also coupled to config](https://github.com/robmorgan/phinx/blob/a601da707f00d40749fe8b2439c28368ed06daed/src/Phinx/Console/Command/Migrate.php#L85). So, we could theoretically hijack config, which is already being passed around between the command and manager, by injecting the dry run setting into config from the command, then pulling that setting value out from the manager to pass to the environment.

That would only alleviate the need to pass the setting separately from the command to the manager, though. Environment is not coupled to config -- it only [receives a migration point and direction](https://github.com/robmorgan/phinx/blob/a601da707f00d40749fe8b2439c28368ed06daed/src/Phinx/Migration/Manager.php#L271) from the manager -- so that approach wouldn't work at that layer.

I'm inclined to say that a [value object](http://shadowhand.me/the-binary-classification-of-objects/) representing the runtime configuration for a migration to be executed might make more sense, to allow for dry run and other options to be supported without elongating parameter lists.